### PR TITLE
Refactor #5139 - leftovers subscribe_to_all_hostgroups

### PR DESCRIPTION
**Pull request by dLobatog**
This functionality is replaced by simply creating a filter "view_hostgroups" , "unlimited" and assigning it to a role. This PR only cleans up the leftovers of it.

### CODE REVIEW

File: app/controllers/hostgroups_controller.rb [hunk 1]
```diff
@@ -44,7 +44,6 @@ def create
     @hostgroup = Hostgroup.new(params[:hostgroup])
     if @hostgroup.save
       # Add the new hostgroup to the user's filters
-      @hostgroup.users << subscribed_users
       @hostgroup.users << users_in_ancestors
```
Comment (domcleal) [inline app/controllers/hostgroups_controller.rb]: Should we remove all of user_hostgroup too?  Is there any use for it?

**Comment by domcleal** [comment]
Maybe the new auth system DB migration should add such a filter?

File: app/controllers/hostgroups_controller.rb (hunk 1 above)
Comment (dLobatog) [inline app/controllers/hostgroups_controller.rb]: I think they've been rendered useless now, @ares any idea if it's being used anywhere?

Comment (ares) [inline app/controllers/hostgroups_controller.rb]: I don't think it's being used anywhere atm

Comment (ohadlevy) [inline app/controllers/hostgroups_controller.rb]: There are a few of those, fact and or etc

Comment (dLobatog) [inline app/controllers/hostgroups_controller.rb]: Right, would you mind if I get rid of them all in this PR?

Comment (domcleal) [inline app/controllers/hostgroups_controller.rb]: Sure, let's have a cleanup.

**Comment by dLobatog** [comment]
[test] because Jenkins hiccup

**Comment by isratrade** [comment]
@eLobato, need to rebase

**Comment by dLobatog** [comment]
@isratrade rebased

**Comment by isratrade** [comment]
looks good. not sure of the test failure, probably jenkins.

**Comment by domcleal** [comment]
Failure of every single test is not likely to be Jenkins' fault.

**Comment by isratrade** [comment]
I didn't test any test failures. It just says "Test Result (no tests)"

**Comment by domcleal** [comment]
Click through all the red circles, then console output and it's failing to db:migrate.

**Comment by dLobatog** [comment]
I've just rebased this again and fixed the errors that made `db:migrate` impossible, which were a typo and a few foreign keys that needed to be removed before dropping the user_xxx tables.

**Comment by dLobatog** [comment]
Rebased again.

**Comment by dLobatog** [comment]
Another rebase

**Comment by dLobatog** [comment]
Latest fix fixes one test that got broken during rebase.

**Comment by domcleal** [comment]
I'm happy with this, but @ohadlevy - could you confirm if there's anything you'll miss here?  (Notices are the only thing that stands out, but they're unused and we have planned to replace them one day.)

File: app/views/api/v1/users/index.json.rabl [hunk 2]
```diff
@@ -1,5 +1,4 @@
 collection @users
 
 attributes :id, :login, :firstname, :lastname, :mail, :admin, :auth_source_id, :role_id, :last_login_on,
-           :domains_andor, :hostgroups_andor, :facts_andor, :filter_on_owner, :compute_resources_andor,
```
Comment (ohadlevy) [inline app/views/api/v1/users/index.json.rabl:4]: any chance this would break anything? should we consider returning empty values instead?

**Comment by ohadlevy** [comment]
@domcleal @eLobato looks good, thanks for cleaning up!

File: app/views/api/v1/users/index.json.rabl (hunk 2 above)
Comment (dLobatog) [inline app/views/api/v1/users/index.json.rabl:4]: After the permission system migration, these values are irrelevant, so it doesn't matter what they return, true/false, they were used to authorize users to domains, hostgroups, and so on, the new system just disregards this. 

If you're using api/v1 and Foreman > 1.5 it'll just mean you'll get default values for these fields, so clients can't really make any use of these. If someone is using these fields in their clients well it's making a big mistake as these fields do not matter at all in Foreman and are never updated according to the authorization system.

Comment (ohadlevy) [inline app/views/api/v1/users/index.json.rabl:4]: while i know they are irrelevant, i'm wondering if its going to break
anyone's automated scripts?

Comment (ohadlevy) [inline app/views/api/v1/users/index.json.rabl:4]: Also.. would you consider doing similar cleanup to javascript/css? ;)

Comment (dLobatog) [inline app/views/api/v1/users/index.json.rabl:4]: Sure, how about we keep api/v1 as is and mention in the docs that these values are not used anywhere and will always return empty strings to keep compatibility?

**Comment by dLobatog** [comment]
Latest update removes '/log' from the repository, `rails s` autocreates it.

File: app/views/api/v1/users/index.json.rabl (hunk 2 above)
Comment (dLobatog) [inline app/views/api/v1/users/index.json.rabl:4]: Although I'm not quite sure where to put that information on the API docs...

**Comment by dLobatog** [comment]
Rebased.

File: app/views/api/v1/users/index.json.rabl (hunk 2 above)
Comment (domcleal) [inline app/views/api/v1/users/index.json.rabl:4]: You could add a "description" bit to the (API) UsersController#show action.

**Comment by dLobatog** [comment]
I rebased this. I thought about adding a description to api/v1/users_controller#show, however removing these values (hostgroups_andor, domains_andor, etc...) from database will result in not returning them in the response even if defined in the RABL. I think that's OK.

**Comment by dLobatog** [comment]
Rebased.

**Comment by isratrade** [comment]
@eLobato, looks good. I suggest to add this commit https://github.com/isratrade/foreman/commit/66303b9bce50f84d31b7b66462f6c83de46413fa to preserve the response for API v1 even if the values are meaningless.

**Comment by dLobatog** [comment]
@isratrade Thanks for your commit, I've put the changes in :smile:

**Comment by dLobatog** [comment]
@isratrade , just rebased this again

**Comment by isratrade** [comment]
@eLobato, failing test on DomainParameterTest

**Comment by dLobatog** [comment]
@isratrade Thanks, I've updated it. :green_apple: now

**Comment by isratrade** [comment]
Merged as 54141ab94976f2c398d94f314cf284fcb88fbc4a. thanks @eLobato !

**Pull request closed without merging**